addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/report/ExtensionAuthhelperReport.java

```diff
@@ -0,0 +1,258 @@
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.addon.authhelper.AuthUtils;
+import org.zaproxy.addon.authhelper.AutoDetectSessionManagementMethodType;
+import org.zaproxy.addon.automation.AutomationEnvironment;
+import org.zaproxy.addon.automation.AutomationPlan;
+import org.zaproxy.addon.automation.AutomationProgress;
+import org.zaproxy.addon.reports.ExtensionReports;
+import org.zaproxy.addon.reports.ReportData;
+import org.zaproxy.zap.authentication.AuthenticationMethod.AuthCheckingStrategy;
+import org.zaproxy.zap.authentication.ManualAuthenticationMethodType.ManualAuthenticationMethod;
+import org.zaproxy.zap.extension.stats.ExtensionStats;
+import org.zaproxy.zap.extension.stats.InMemoryStats;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.model.SessionStructure;
+
+public class ExtensionAuthhelperReport extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionAuthhelperReport";
+
+    private static final List<Class<? extends Extension>> DEPENDENCIES =
+            List.of(ExtensionReports.class);
+    private static final Logger LOGGER = LogManager.getLogger(ExtensionAuthhelperReport.class);
+
+    private AuthReportDataHandler authReportDataHandler;
+
+    public ExtensionAuthhelperReport() {
+        super(NAME);
+    }
+
+    @Override
+    public boolean supportsDb(String type) {
+        return true;
+    }
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+    }
+
+    @Override
+    public void optionsLoaded() {
+        ExtensionReports extReports = AuthUtils.getExtension(ExtensionReports.class);
+        if (extReports != null) {
+            LOGGER.debug("Loaded client script based auth type.");
+            authReportDataHandler = new AuthReportDataHandler();
+            extReports.addReportDataHandler(authReportDataHandler);
+        }
+    }
+
+    @Override
+    public void unload() {
+        ExtensionReports extReports = AuthUtils.getExtension(ExtensionReports.class);
+        if (extReports != null && authReportDataHandler != null) {
+            extReports.removeReportDataHandler(authReportDataHandler);
+        }
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("authhelper.authreport.desc");
+    }
+
+    @Override
+    public String getUIName() {
+        return Constant.messages.getString("authhelper.authreport.name");
+    }
+
+    private class AuthReportDataHandler implements ExtensionReports.ReportDataHandler {
+
+        @Override
+        public void handle(ReportData reportData) {
```

Comment:
Maybe I'm over thinking it, but sonar would definitely complain about this being a "brain" method 🧠

Reply:
What am I supposed to do about that? 😁

Reply:
Do you think that will make it easier to follow? 
TBH I dont care about what sonar says, but I do care about what the other people working on this code think 😁

Reply:
Fair, IMHO it would definitely be easier to work with if it was split up IMHO. As a single monolithic method it's hard(er) to work with.

Reply:
I've added a helper method which will reduces the size a bit and hopefully makes it a bit clearer.